src/langchain/base_memory.ts

```diff
@@ -0,0 +1,110 @@
+import {
+   BaseMemory,
+   InputValues,
+   OutputValues,
+   getInputValue,
+   getOutputValue,
+} from "@langchain/core/memory";
+import {
+   BaseChatMessageHistory,
+   BaseListChatMessageHistory,
+} from "@langchain/core/chat_history";
+import { BaseMessage } from "@langchain/core/messages";
+
+/**
+ * Class for storing chat message history in-memory. It extends the
+ * BaseListChatMessageHistory class and provides methods to get, add, and
+ * clear messages.
+ */
```

Comment:
@danielchalef I need to set up base memory here in order to avoid listing non-core langchain package as a dependency